modules/tinymce/src/core/test/ts/browser/util/ImageUploaderTest.ts

```diff
@@ -0,0 +1,59 @@
+import { Log, Pipeline, Chain } from '@ephox/agar';
+import { Assert, UnitTest } from '@ephox/bedrock-client';
+import { ApiChains, TinyLoader } from '@ephox/mcagar';
+import { Arr } from '@ephox/katamari';
+import ImageUploader, { UploadResult } from 'tinymce/core/api/util/ImageUploader';
+
+import { BlobCache, BlobInfo } from 'tinymce/core/api/file/BlobCache';
+
+import Theme from 'tinymce/themes/silver/Theme';
+
+UnitTest.asynctest('browser.tinymce.core.util.ImageUploaderTest', (success, failure) => {
+
+  Theme();
+
+  TinyLoader.setupLight((editor, onSuccess, onFailure) => {
+    const cache = BlobCache();
+    const blob = new Blob([ JSON.stringify({ hello: 'world' }) ]);
+
+    const image1 = cache.create({ blob, base64: 'test' });
+
+    const cUploadImages = (images: BlobInfo[], notifications: boolean) => Chain.async((_value, next, die) => {
+      const uploader = ImageUploader(editor);
+      uploader.upload(images, notifications).then(next, die);
+    });
+
+    const cAssertUploadResultSuccess = (expectedLength: number, url: string) => Chain.op((uploadResults: UploadResult[]) => {
+      Assert.eq('Number of results', expectedLength, uploadResults.length);
+      Arr.each(uploadResults, (uploadResult) => {
+        Assert.eq('Url is Image.png', url, uploadResult.url);
+        Assert.eq('Successful', true, uploadResult.status);
+      });
+    });
+
+    const cAssertUploadResultFailure = (expectedLength: number, url: string) => Chain.op((uploadResults: UploadResult[]) => {
+      Assert.eq('Number of results', expectedLength, 0);
+      Arr.each(uploadResults, (uploadResult) => {
```

Comment:
If the expected length of the results is 0, then `Arr.each` will not run?

Reply:
No, it's actually getting there.

![Screen Shot 2020-11-13 at 8 48 49 am](https://user-images.githubusercontent.com/6000860/99005764-210c4200-258d-11eb-8f56-4e640e47426e.png)

![Screen Shot 2020-11-13 at 8 48 56 am](https://user-images.githubusercontent.com/6000860/99005767-22d60580-258d-11eb-9fc3-8bed3b1f9cf2.png)

Reply:
Yes, but if the expected length of the results is 0, as the line above intends to assert, then this would not run.

Reply:
Please see comments above